crates/compiler/fmt/tests/test_fmt.rs

```diff
@@ -5705,6 +5705,39 @@ mod test_fmt {
         ));
     }
 
+    #[test]
+    fn clauses_with_multiple_abilities() {
+        expr_formats_same(indoc!(
+            r#"
+            f : {} -> a | a has Eq & Hash & Decode
+
+            f
+            "#
+        ));
+
+        expr_formats_to(
+            indoc!(
+                r#"
+                f : {} -> a | a has Eq & Hash & Decode,
+                              b has Eq & Hash
+
+                f
+                "#
+            ),
+            indoc!(
+                // TODO: ideally, this would look a bit nicer - consider
+                // f : {} -> a
+                //   | a has Eq & Hash & Decode,
+                //     b has Eq & Hash
```

Comment:
In a case like this, I do think `where` could look nicer, e.g.

```elm
f : {} -> a
    where
        a has Eq & Hash & Decode,
        b has Eq & Hash
```

Reply:
With consistent prefix punctuation, this could look pretty clean with `|` and `,` with the following format:

```
f : {} -> a
  | a has Eq & Hash & Decode
  , b has Eq & Hash
```

(at least from my perspective, mixing prefix and suffix punctuation/tokens should be avoided)

Reply:
Actually, per @ayazhafiz point, there's a Zulip thread for this, so I'll repost there